### DISCUSSION TITLE
Memory requirements for large app

### DISCUSSION THREAD

**Discussion by safwanadnan** [devlikeapro/waha#572] [Q&A] [answered]
I am building an app and as of now my waha instance is running on a vps with 4 vCPU cores, 6 GB RAM and 100 GB NVMe SSD storage. I store only session data and no media. As of now I am in developing the app but when it is deployed I will have well over 1000 instances running. What will be recommended memory requirements for it to function smoothly considering active use regularly. Also is x GB of NVMe storage better or 4x GB of SSD for the same price. Also should i stick to the $19 plan or will getting the $100 plan make a difference?

**Comment by devlikepro** [comment, marked as answer]
Hi! It depends on what engine you're going to use - WEBJS (runs real browser) or NOWEB (works with websocket connections)
Here's answer about how many CPU/Memory you should have for that, based on # of sessions you'll have
https://waha.devlike.pro/docs/overview/faq/#how-much-cpu-and-memory-does-waha-consume

I guess it'll be NOWEB, so anyway consider running TWO containers with ~500 sessions (or even split them into more smaller one), this way you'll avoid problems when one session goes crazy and affects all sessions at once (it doesn't happen frequently, but shit happens)

Here's an article for that:
[**WAHA Scaling - How To Handle 500+ Sessions**](https://waha.devlike.pro/blog/waha-scaling-how-to-handle-500-sessions/)

We're also working on **multiple workers on 1 database** approach, will release it at the end of October :crossed_fingers: 

> Also should i stick to the $19 plan or will getting the $100 plan make a difference?

Here's the differences - https://waha.devlike.pro/pricing/
if you have 100+ sessions - consider getting PRO ($99), this way we can help you more personally in case of issues :+1: 


[![patron:PRO](https://img.shields.io/badge/patron-PRO-188a42)](https://waha.devlike.pro/docs/how-to/plus-version/#tiers)

**Reply by safwanadnan** [reply]
Also I wanted to enquire about auto updates. How do I do it and are you sure it won't disrupt the instances running in any way?

**Reply by devlikepro** [reply]
It'll stop and start sessions again if you set `WHATSAPP_RESTART_ALL_SESSIONS=True` flag
https://waha.devlike.pro/docs/how-to/config/#sessions
Consider updating docker image during the nights, so no communication happens during that time

[![patron:PRO](https://img.shields.io/badge/patron-PRO-188a42)](https://waha.devlike.pro/docs/how-to/plus-version/#tiers)

**Reply by safwanadnan** [reply]
My app has international usage so the time zone won't make much of a difference. However, how long does it typically take to update. It will still function normally while I pull the docker image if I'm not wrong? And then I can stop and start my image again? What if I donot set the restart session flag? Will all my instances be logged out or something?

**Reply by FaheemAlvii** [reply]
based on specifications you mentioned I think you are considering Contabo.. if so then make sure to use atleast 2 nodes (use load balancer) in different regions as contabo is not very good sometimes..

**Reply by safwanadnan** [reply]
Yes I am on Contabo and it's disappointing to know that after configuring everything it might not be good enough. What are other alternatives in similar price range?

**Reply by safwanadnan** [reply]
I sometimes get a 500 internal server error and that's with just one instance running. I fixed it with exception handling and retry logic but I reckon it would pose a problem when the application is live. Do you think that's the Contabo going haywire? Also for multiple nodes do I need to purchase multiple waha images or one will work fine?

**Reply by FaheemAlvii** [reply]
> Yes I am on Contabo and it's disappointing to know that after configuring everything it might not be good enough. What are other alternatives in similar price range?

Hetner... it gives u a bit less specs than contabo but atleast more stable

**Reply by safwanadnan** [reply]
what about netcup? any reviews? it's cheaper than Contabo as well surprisingly

**Reply by FaheemAlvii** [reply]
> what about netcup? any reviews? it's cheaper than Contabo as well surprisingly

Only issue with netcup is bad support... Let's keep the discussion relevant and if you have server related questions you can ask me on Discord: "_cyborg_" with underscore in start and end
